.github/workflows/generate-kubelet-flags.yaml

```diff
@@ -9,6 +9,27 @@ jobs:
     - name: Set up containerd
       uses: crazy-max/ghaction-setup-containerd@v2
     - uses: actions/checkout@v3
-    - run: |
+    - name: Set up branch for kubelet flag changes
+      run: |
+        TIMESTAMP=$(date -d "${{ github.event.head_commit.timestamp }}" +'%Y-%m-%d-%H-%M-%S')
+        echo "TIMESTAMP=$TIMESTAMP" >> $GITHUB_ENV
+        git checkout -b kubeletFlags/$TIMESTAMP
+    - name: Generate valid kubelet flags per supported k8s versions
+      run: |
         make generate-kubelet-flags
-      name: Generate valid kubelet flags per supported k8s versions
\ No newline at end of file
+    - name: Create PR for modified kubelet flag files
+      run: |
+        git config --global user.email "aleldeib@microsoft.com"
+        git config --global user.name "alexeldeib"
+        git remote set-url origin https://alexeldeib:$github_token@github.com/Azure/AgentBaker.git
+        git add .
+        git commit -m"feat: generate kubelet flags for supported versions in manifest.json"
+        git push -u origin kubeletFlags/$timestamp -f
+        curl \
+          -X POST \
+          https://api.github.com/repos/Azure/AgentBaker/pulls \
+          -d '{"head" : "'kubeletFlags/$timestamp'", "base" : "master", "title" : "bump: automated PR for updating kubelet flags files"}' \
+          -u "alexeldeib:$github_token"
+      env:
+        github_token: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        timestamp: ${{ env.TIMESTAMP }}
```

Comment:
is this not already in env?

Reply:
no, I have to pipe it down from the previous stage where I parsed it.

Reply:
but you don't need to reset timestamp from env.TIMESTAMP, right? 

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable

Reply:
resolved